uaclient/util.py

```diff
@@ -293,12 +293,18 @@ def get_dict_deltas(
 
 
 @lru_cache(maxsize=None)
-def get_machine_id(data_dir: str) -> str:
+def get_machine_id(cfg) -> str:
```

Comment:
Let's move this function into uaclient.contract since that's our only callsite, It doesn't make sense to pollute uaclient.util with this function and cope with nested import issues.
Also let's use typing on everything we touch.... 
so this could be either 
```suggestion
def get_machine_id(cfg: UAConfig) -> str:
```


Or maybe we don't even need to pass in cfg, but instead just accept a fallback_machine_id_file
which can be obtained at the call_site with get_machine_id(self.cfg.data_path("machine-id"))

Reply:
About moving it to `uaclient.contract`, my only question would be if we have any expectation of using it somewhere else. Getting the machine ID seems generic enough imho, but if we will keep using it only in contracts, then I agree that it makes sense.